wifiphisher/common/victim.py

```diff
@@ -0,0 +1,159 @@
+"""
+This module was made to keep track the information
+of the victims connected to the rogue AP
+"""
+
+import os
+from macmatcher import MACMatcher as macmatcher
+import wifiphisher.common.constants as constants
+
+
+class Victim(object):
+    """
+    Resembles a Victim (i.e. a connected device to the rogue AP)
+    """
+
+    def __init__(self, vmac_address, ip_address):
+        self.vmac_address = vmac_address
+        self.ip_address = ip_address
+        self.os = "Unknown"
+        self.vendor = "Unknown"
+
+    def associate_victim_mac_to_vendor(self, vmac_address):
+        """
+        receives a victims mac address as input, finds the corresponding vendor
+        by using a macmacther object and then accesses the victim
+        dictionary and changes the vendor for the victim with the
+        corresponding mac address
+
+        :param self: empty Victim instance
+        :type self: Victim
+        :param vmac_address: mac address of the victim
+        :type vmac_address: str
+        """
+
+        path_to_mac_file = constants.MAC_PREFIX_FILE
+        macmacther_instance = macmatcher(path_to_mac_file)
+        vendor = macmacther_instance.get_vendor_name(vmac_address)
+        Victims_instance = Victims.get_instance()
+        if vmac_address in Victims_instance.victims_dic:
+            Victims_instance.victims_dic[vmac_address].vendor = vendor
+        else:
+            raise Exception("Error: No such mac address exists in dictionary")
+
+    def assign_ip_to_victim(self, vmac_address, ip_address):
+        """
+        Updates the ip address of the victim with the corresponding mac address
+        """
+
+        Victims_instance = Victims.get_instance()
+        if vmac_address in Victims_instance.victims_dic:
+            Victims_instance.victims_dic[vmac_address].ip_address = ip_address
+        else:
+            raise Exception("Error: No such mac address exists in dictionary")
+
+
+class Victims():
+    """
+    Singleton class that manages all of the victims
+    """
+
+    # Instance will be stored here.
+    __instance = None
+
+    @staticmethod
+    def get_instance():
+        if Victims.__instance is None:
+            Victims()
+        return Victims.__instance
+
+    def __init__(self):
+        if Victims.__instance is not None:
+            raise Exception("Error: Victims class is a singleton!")
+        else:
+            Victims.__instance = self
+            self.victims_dic = {}
+
+    def add_to_victim_dic(self, Victim):
+        """
+        Add new victim to the dictionary
+        """
+
+        self.victims_dic[Victim.vmac_address] = Victim
+
+    def get_print_representation(self):
+        """
+        Returns the victim dictionary in str format in order to be printed
+
+        :param self: Victims instance
```

Comment:
Instead of these paramater information it would be great if you could add function type annotations like [this](https://mypy.readthedocs.io/en/stable/cheat_sheet.html).